From linux-usb-users-bounces@lists.sourceforge.net Sat Feb  3 19:00:10 2007
From: Al Viro <viro@ftp.linux.org.uk>
Date: Sun, 4 Feb 2007 02:59:24 +0000
Subject: USB: fix rtl8150
To: torvald@linux-foundation.org
Cc: linux-kernel@vger.kernel.org, linux-usb-users@lists.sourceforge.net
Message-ID: <20070204025924.GI10050@ftp.linux.org.uk>
Content-Disposition: inline


That code doesn't do what its author apparently thought
it would do...

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/usb/net/rtl8150.c
+++ b/drivers/usb/net/rtl8150.c
@@ -284,7 +284,8 @@ static int write_mii_word(rtl8150_t * dev, u8 phy, __u8 indx, u16 reg)
 	u8 data[3], tmp;
 
 	data[0] = phy;
-	*(data + 1) = cpu_to_le16p(&reg);
+	data[1] = reg & 0xff;
+	data[2] = (reg >> 8) & 0xff;
 	tmp = indx | PHY_WRITE | PHY_GO;
 	i = 0;
 

-------------------------------------------------------------------------
Using Tomcat but need to do more? Need to support web services, security?
Get stuff done quickly with pre-integrated technology to make your job easier.
Download IBM WebSphere Application Server v.1.0.1 based on Apache Geronimo
http://sel.as-us.falkag.net/sel?cmd=lnk&kid=120709&bid=263057&dat=121642
_______________________________________________
Linux-usb-users@lists.sourceforge.net
To unsubscribe, use the last form field at:
https://lists.sourceforge.net/lists/listinfo/linux-usb-users
